net: ieee802154_radio: Enable ed_scan unconditionally

rlubos opened the pull request:
Energy scan procedure, while introduced specifically for OpenThread in
Zephyr, may also be used by other upper layers (like Zigbee).
Therefore, disable conditional inclusion of the `ed_scan` API.

Additionally, add stub functions in radio drivers that do not implement
this API yet.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
FYI @MarekPorwisz @mariuszpos

tbursztyka commented on include/net/ieee802154_radio.h:
```diff
@@ -157,15 +157,13 @@ struct ieee802154_radio_api {
 	/** Run an energy detection scan.
-	 * Note: channel must be set prior to request this function.
-	 * duration parameter is in ms.
+	 *  Note: channel must be set prior to request this function.
```
why the additional space?

rlubos replied:
I just thought it looks nicer when aligned, changed w/o thinking about it too much. I can revert this change if you prefer it as it was.

jukkar commented on include/net/ieee802154_radio.h:
```diff
@@ -157,15 +157,13 @@ struct ieee802154_radio_api {
+	 *  Note: channel must be set prior to request this function.
+	 *  duration parameter is in ms.
 	 */
 	int (*ed_scan)(struct device *dev,
```
Instead of adding dummy ed_scan API function to all the drivers, could we check whether the ed_scan pointer is set before calling it. This way we do not really need to add these dummy functions to all drivers. What do you think?

rlubos replied:
Sure, sounds very reasonable, will fix.

rlubos replied:
Removed stubs, added a check in the OT radio layer.

rlubos committed:
net: ieee802154_radio: Enable ed_scan unconditionally

Energy scan procedure, while introduced specifically for OpenThread in
Zephyr, may also be used by other upper layers (like Zigbee).
Therefore, disable conditional inclusion of the `ed_scan` API.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: openthread: Check if ed_scan API is implemented before use

Verify that `ed_scan` is implemented by the radio driver before use. In
case it's not, return appropriate error code to OpenThread.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
